core/src/main/java/org/apache/carbondata/core/util/DataTypeUtil.java

```diff
@@ -331,6 +331,10 @@ public static Object getDataBasedOnDataType(String data, DataType actualDataType
       return ByteUtil.toBytes(Integer.parseInt(dimensionValue));
     } else if (actualDataType == DataTypes.LONG) {
       return ByteUtil.toBytes(Long.parseLong(dimensionValue));
+    } else if (actualDataType == DataTypes.DOUBLE) {
+      return ByteUtil.toBytes(Double.doubleToLongBits(Double.parseDouble(dimensionValue)));
```

Comment:
Add methods in ByteUtil to convert from double to byte array and byte array to double instead of handling here